### PR TITLE
add `ambiguity_threshold` param to `FallbackClassifier`

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/6244

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation (part of https://github.com/RasaHQ/rasa/pull/6195)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/classifiers/fallback_classifier.py [hunk 1]
```diff
@@ -47,10 +54,55 @@ def process(self, message: Message, **kwargs: Any) -> None:
...
+    def _nlu_confidence_below_threshold(self, message: Message) -> Tuple[bool, float]:
+        nlu_confidence = message.data[INTENT].get(INTENT_CONFIDENCE_KEY)
+        return nlu_confidence < self.component_config[THRESHOLD_KEY], nlu_confidence
+
+    def _nlu_prediction_ambiguous(
+        self, message: Message
+    ) -> Tuple[bool, Optional[float]]:
+        intents = message.data[INTENT_RANKING_KEY]
+        if len(intents) >= 2:
+            first_confidence = intents[0].get("confidence", 1.0)
```
Comment (tabergma) [inline rasa/nlu/classifiers/fallback_classifier.py:98]:
```suggestion
            first_confidence = intents[0].get(INTENT_CONFIDENCE_KEY, 1.0)
```

File: rasa/nlu/classifiers/fallback_classifier.py [hunk 2]
```diff
@@ -47,10 +54,55 @@ def process(self, message: Message, **kwargs: Any) -> None:
...
+            second_confidence = intents[1].get("confidence", 1.0)
```
Comment (tabergma) [inline rasa/nlu/classifiers/fallback_classifier.py:99]:
```suggestion
            second_confidence = intents[1].get(INTENT_CONFIDENCE_KEY, 1.0)
```

File: rasa/nlu/classifiers/fallback_classifier.py [hunk 3]
```diff
@@ -47,10 +54,55 @@ def process(self, message: Message, **kwargs: Any) -> None:
...
-        )
+        """Check if the fallback intent should be predicted.
+
+        Args:
+            message: The current message and its intent predictions.
+
+        Returns:
+            `True` if the fallback intent should be predicted.
+        """
+        intent_name = message.data[INTENT].get("name")
```
Comment (tabergma) [inline rasa/nlu/classifiers/fallback_classifier.py:65]: Minor: As we have a constant for "confidence" does it make sense to also introduce one for "name"?

Comment (wochinge) [reply rasa/nlu/classifiers/fallback_classifier.py]: done and tried to replace it in most files

**Pull request merged**